playground/src/main/java/org/navgurukul/playground/ui/PlaygroundActivity.kt

```diff
@@ -70,19 +75,40 @@ class PlaygroundActivity : AppCompatActivity() {
         }
     }
 
+    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
+        val inflater: MenuInflater = menuInflater
+        inflater.inflate(R.menu.playground_menu, menu)
+        return true
+    }
+
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
         val id: Int = item.itemId
         if (id == android.R.id.home) {
             // If home button pressed Exit the activity
             super.onBackPressed()
             return true
         }
+        else if(id == R.id.clear){
+            etCode.text.clear()
+        }
         return super.onOptionsItemSelected(item)
     }
 
+    override fun onBackPressed() {
+        if(sheetBehavior.state == BottomSheetBehavior.STATE_EXPANDED){
+            // If bottom sheet is expanded, collapse it on back button
+            sheetBehavior.state = BottomSheetBehavior.STATE_COLLAPSED
+            return
+        }
+        super.onBackPressed()
+    }
+
     private fun createCode() {
         etCode = findViewById(R.id.etCode)
 
+        // Restore code from shared Prefs
+        etCode.setText(sharedPreferences.getString(KEY_PREF_CODE_BACKUP,""))
```

Comment:
same here, should be delegated via viewModel

Reply:
Done